frontend/src/features/public-form/components/FormPaymentPage/stripe/utils.ts

```diff
@@ -0,0 +1,16 @@
+import { PaymentIntent } from '@stripe/stripe-js'
+
+export enum PaymentViewStates {
+  Invalid,
+  Succeeded,
+  Canceled,
+  Pending,
+}
+export const getPaymentViewStates = (
+  status: PaymentIntent.Status | undefined,
+): PaymentViewStates => {
+  if (!status) return PaymentViewStates.Invalid
+  if (['succeeded'].includes(status)) return PaymentViewStates.Succeeded
+  if (['canceled'].includes(status)) return PaymentViewStates.Canceled
+  return PaymentViewStates.Pending
+}
```

Comment:
i feel like for safety's sake, we should just handle it (since one of the valid states for payment_intent is processing). nonetheless, doesn't have to be done now. we can open a ticket and deal with it later